python_modules/dagster/dagster/_core/definitions/asset_check_spec.py

```diff
@@ -31,23 +33,26 @@ class AssetCheckSeverity(Enum):
     ERROR = "ERROR"
 
 
-class AssetCheckSpec(
-    NamedTuple(
-        "_AssetCheckSpec",
-        [
-            ("name", PublicAttr[str]),
-            ("asset_key", PublicAttr[AssetKey]),
-            ("description", PublicAttr[Optional[str]]),
-            ("additional_deps", PublicAttr[Iterable["AssetDep"]]),
-            (
-                "blocking",  # intentionally not public, see https://github.com/dagster-io/dagster/issues/20659
-                bool,
-            ),
-            ("metadata", PublicAttr[Optional[Mapping[str, Any]]]),
-            ("automation_condition", Optional["AutomationCondition[AssetCheckKey]"]),
-        ],
+LazyAutomationCondition: TypeAlias = Annotated[
+    "AutomationCondition",
```

Comment:
leave a comment that this would ideally be `AutomationCondition[AssetCheckKey]` if record was updated to handle it

Reply:
ha diamond failed me this time